core/src/main/scala/io/finch/Input.scala

```diff
@@ -67,29 +67,14 @@ final case class Input(request: Request, route: List[String]) {
    */
   def withForm(params: (String, String)*): Input = {
     // TODO: Figure out way to do that w/o Netty.
-    val dataFactory = new DefaultHttpDataFactory(false) // we don't use disk
-    val encoder = new HttpPostRequestEncoder(
-      dataFactory,
-      new DefaultHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.POST, request.path),
-      false
-    )
-
-    params.foreach {
-      case (k, v) => encoder.addBodyAttribute(k, v)
-    }
-
-    val req = encoder.finalizeRequest()
-
-    val content = if (req.isChunked) {
-      Buf(
-        Iterator.continually(encoder.nextChunk())
-        .takeWhile(c => !c.isLast)
-        .map(c => ChannelBufferBuf.Owned(c.getContent))
-        .toVector
-      )
-    } else ChannelBufferBuf.Owned(req.getContent)
-
-    withBody[Application.WwwFormUrlencoded](content, StandardCharsets.UTF_8)
+     val postRequest: Request = RequestBuilder()
+       .setHeader(Fields.UserAgent, "MyBot")
+       .setHeader(Fields.Connection, HttpHeaders.Values.KEEP_ALIVE)
+       .addFormElement(params: _*)
+       .url("http://foo")
```

Comment:
this is extremely regrettable as we can't construct a Request without a url. I tried a lot of other ways to accomplish this same goal of producing a buf from a form encoded request but I'm at a loss as to any way to do it cleanly, while relying on Finagle code as much as possible that doesn't involve setting a bogus url on the request

Reply:
Oh yea, that's right. Maybe just use `http://localhost` instead?